src/realtimecontroller/RealtimeState.ts

```diff
@@ -83,4 +84,18 @@ export default class RealtimeState {
    * Callbacks to listen for fatal errors
    */
   fatalErrorCallbacks: ((error: Error) => void)[] = [];
+
+  /**
+   * Callbacks to trigger when sending message
+   */
+  sendDataMessageCallbacks: ((
+    topic: string, // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    data: Uint8Array | string | any,
+    lifetimeMs?: number
+  ) => void)[] = [];
+
+  /**
+   * Callbacks to listen for receiving message from data channel based on given topic
+   */
+  didReceiveDataMessageCallbacks: Map<string, ((dataMessage: DataMessage) => void)[]> = new Map();
```

Comment:
nit: only this method starts with `did`. Exising properties do not start with a verb.